packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -634,4 +634,27 @@ - (void)setMixWithOthers:(FLTMixWithOthersMessage *)input
   }
 }
 
+- (void)setAmbientMix:(FLTAmbientMixMessage *)input
+                error:(FlutterError *_Nullable __autoreleasing *)error {
+  AVAudioSessionCategory sessionCategory =
+      [input.ambientMix boolValue] ? AVAudioSessionCategoryAmbient : AVAudioSessionCategoryPlayback;
+  AVAudioSessionCategoryOptions categoryOptions = [input.mixWithOthers boolValue]
+                                                      ? AVAudioSessionCategoryOptionMixWithOthers
+                                                      : AVAudioSessionCategoryOptionDuckOthers;
+
+  if (@available(iOS 12.0, *)) {
+    AVAudioSessionMode sessionMode =
+        [input.ambientMix boolValue] ? AVAudioSessionModeVoicePrompt : AVAudioSessionModeDefault;
```

Comment:
can you explain why choose `voicePrompt` instead of so many other options? Will it be better to pass an enum directly instead of a boolean? 

same question for `sessionCategory` and `categoryOptions`

Reply:
Sure, this combination is suitable for allowing all audio sources within the app and other apps to mix together. Based on my experiment, this combination works best.

But you are absolutely right, it should accept an enum and the client should have control over this.

I think what **.setAmbientMix(...)** does is very app specific and not suitable video_player.